### PR TITLE
Strip out "spiffe://" in the identity

### PR DESCRIPTION
**Pull request by JimmyCYJ**
**What this PR does / why we need it**:Strips out the prefix "spiffe://" from the URI in the SAN field of the peer certificate.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by kyessenov** [comment]
What's the motivation?

File: src/envoy/mixer/utils.cc [hunk 1]
```diff
@@ -54,8 +54,14 @@ bool GetSourceUser(const Network::Connection* connection, std::string* user) {
   if (connection) {
     Ssl::Connection* ssl = const_cast<Ssl::Connection*>(connection->ssl());
     if (ssl != nullptr) {
-      *user = ssl->uriSanPeerCertificate();
-      return true;
+      std::string result = ssl->uriSanPeerCertificate();
```
Comment (wattli) [inline src/envoy/mixer/utils.cc]: Is there any test we need to update as well?

Comment (JimmyCYJ) [reply src/envoy/mixer/utils.cc]: I have run all existing tests in /proxy via bazel test ..., and all of them pass.

File: src/envoy/mixer/utils.cc [hunk 2]
```diff
@@ -54,8 +54,14 @@ bool GetSourceUser(const Network::Connection* connection, std::string* user) {
...
+      std::string prefix = "spiffe://";
```
Comment (qiwzhang) [inline src/envoy/mixer/utils.cc]: define it as global const  in annoymous namespace

File: src/envoy/mixer/utils.cc [hunk 3]
```diff
@@ -54,8 +54,14 @@ bool GetSourceUser(const Network::Connection* connection, std::string* user) {
...
+      // Strip out the prefix "spiffe://" in the identity.
+      std::size_t found = result.find(prefix);
```
Comment (qiwzhang) [inline src/envoy/mixer/utils.cc]: use compare like this

https://github.com/istio/mixerclient/blob/master/src/attributes_builder.cc#L34

find() is very expensive

File: src/envoy/mixer/utils.cc [hunk 4]
```diff
@@ -54,8 +60,13 @@ bool GetSourceUser(const Network::Connection* connection, std::string* user) {
...
     if (ssl != nullptr) {
-      *user = ssl->uriSanPeerCertificate();
-      return true;
+      std::string result = ssl->uriSanPeerCertificate();
+      if (result.length() >= kSPIFFEPrefix.length() &&
+          result.compare(0, kSPIFFEPrefix.length(), kSPIFFEPrefix) == 0) {
+        // Strip out the prefix "spiffe://" in the identity.
+        *user = result.substr(kSPIFFEPrefix.size());
+        return true;
+      }
```
Comment (qiwzhang) [inline src/envoy/mixer/utils.cc]: what about else? the one doesn't match the prefix?

Comment (wattli) [reply src/envoy/mixer/utils.cc]: if not, we don't need to do anything. So far, all id will have this prefix.

Comment (qiwzhang) [reply src/envoy/mixer/utils.cc]: I think we still want to get source.user even it doesn't match spiff prefix.

Comment (qiwzhang) [reply src/envoy/mixer/utils.cc]: what format will it look like?   will it have similar format of xyz://  If so,  we want to strip it too.

Comment (wattli) [reply src/envoy/mixer/utils.cc]: Good point, as mentioned, so far, all Ids have prefix "spiffe://", i don't think we have plan to support others in the near future.

Comment (JimmyCYJ) [reply src/envoy/mixer/utils.cc]: From offline discussion with Wayne, if there is no "spiffe://" prefix, assign the uri to user and return true. This helps to keep consistent behavior for cases where uri does not begin with "spiffe://".

**Pull request merged**